addons/website_sale/static/src/js/website_sale_recently_viewed.js

```diff
@@ -56,31 +56,32 @@ publicWidget.registry.productsRecentlyViewedSnippet = publicWidget.Widget.extend
             return;
         }
         var products = res['products'];
-        if (products && products.length) {
```

Comment:
I would keep the condition this way and add an `else` for the hiding part

Reply:
I have to render it when there is no product. That was a bug, the last product stayed in the carousel when added to cart .